### ISSUE TITLE
Not compatible with Deno 1.21

### ISSUE THREAD

**Issue by deckchairlabs** [shimataro/node-adjuster#555]
The ValueSchemaError is not type compatible with at least Deno 1.21

```
TS2416 [ERROR]: Property 'cause' in type 'ValueSchemaError' is not assignable to the same property in base type 'Error'
```

**Comment by shimataro** [comment]
@deckchairlabs 
Thank you for creating issue! and sorry for late response...

[A new property `cause` has been added to `Error`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) in ES2022, and it seems to conflict with `ValueSchemaError.cause`.

<https://github.com/shimataro/value-schema/blob/a9229368891be8416b2d901af749dc67363cfb91/src/libs/ValueSchemaError.ts#L28>

`ValueSchemaError.cause` needs to be renamed, I will deal with it in v4 since it will be breaking change.

**Comment by shimataro** [comment]
@deckchairlabs 
I have released [v4.0.0-rc.1](https://deno.land/x/value_schema@v4.0.0-rc.1), that supports Deno>=1.20.
Please check [migration guide](https://github.com/shimataro/value-schema/blob/v4/migration.md)!
